src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -508,7 +531,16 @@ namespace cryptonote
     m_blockchain_storage.set_user_options(blocks_threads,
         blocks_per_sync, sync_mode, fast_sync);
 
-    r = m_blockchain_storage.init(db.release(), m_nettype, m_offline, test_options);
+    const std::pair<uint8_t, uint64_t> regtest_hard_forks[3] = {std::make_pair(1, 0), std::make_pair(7, 1), std::make_pair(0, 0)};
```

Comment:
That's going to be forgotten when we switch to v8, etc. If you want to set it to the last fork, request it from the hard fork object (get_ideal_version).

Reply:
Do you mean move the logic to ```cryptonote_core.cpp``` to ensure it's close to the other hard_fork structs and it won't be forgotten? I don't see how ```get_ideal_version``` comes into play here, as for the regtest/FAKECHAIN nettype hardforks still need to be set for a low height...

Reply:
@moneromooo-monero 
The function `get_ideal_version` assumes that the internal hard fork heights info is already initialized, which isn't the case here. Also, in this case we want to get the hard fork info heights info of MAINNET while the network type is set to FAKECHAIN. I've made a patch https://github.com/stoffu/monero/commit/a758f1dc7a878f086e36b5101fe0987e548a9213 that makes `HardFork:: Params` a public class and adds a static function `Blockchain::get_hard_fork_heights(network_type nettype)`.

Reply:
Ah, yes, chicken and egg. OK. Somehow it seems like we're going about this the wrong way, but I can't quite put my finger on it.

Reply:
Updated with stoffu's adjustments. I've tried thinking about what would be wrong about the initialization of regtest_hard_forks, but can't seem to identify the chicken and egg situation here...